### PR TITLE
domain: make the transaction from `initStatsCtx` blocking gc

### LINKED ISSUE #53592: min start ts of `initStatsCtx` is not tracked
## Bug Report

Please answer these questions before submitting your issue. Thanks!

https://github.com/pingcap/tidb/blob/2300ac6fe6cb29de119ccc04478ffaf4df463ffc/pkg/session/session.go#L3510-L3514

The min start ts will block gc to guarantee the ongoing transaction can see its data, but the min start ts of `initStatsCtx` is not tracked.

### 1. Minimal reproduce step (Required)

Inject 30min sleep when loading stats in init context.

See [commit](https://github.com/you06/tidb/commit/e01327d73b38a38be08d45fc8cd33ac316a70731).

### 2. What did you expect to see? (Required)

Stats loaded after 30min.

### 3. What did you see instead (Required)

```log
[2024/05/27 21:08:25.631 +09:00] [ERROR] [domain.go:2314] ["init stats info failed"] [lite=true] ["take time"=30m0.002233833s] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2024-05-27 20:38:25.628 +0900 JST, GC safe point is 2024-05-27 20:49:25.682 +0900 JST"]
```

### 4. What is your TiDB version? (Required)

f5ac93e448a

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve?

Issue Number: close #53592

Problem Summary:

The `initStatsCtx` is not get from internal session pool, so it's not tracked by min start ts service, thus gc lifetime can overtake it easily.

### What changed and how does it work?

Insert it into tracked session table and it will block gc.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

I reproduce the test mentioned in issue, the stats are loaded successfully after 30min.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix a bug that tidb will report gc lifetime too short when tidb bootstrap with slow loading statistics
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: pkg/domain/domain.go [hunk 1]
```diff
@@ -2256,6 +2256,22 @@ func (do *Domain) UpdateTableStatsLoop(ctx, initStatsCtx sessionctx.Context) err
 		},
 		"analyzeJobsCleanupWorker",
 	)
+	go func() {
```
Comment (hawkingrei) [inline pkg/domain/domain.go:2259]: Can you add test for it ?

Comment (hawkingrei) [inline pkg/domain/domain.go:2259]: do.wg.Run(func(){
},"DeleteInternalSession init stats")

**Review by qw4990 — APPROVED** [review]

File: pkg/domain/domain.go (hunk 1 above)
Comment (qw4990) [reply pkg/domain/domain.go:2259]: +1, how to test this?

**Review by hawkingrei — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Pull request merged**